src/sonar_widget/SonarPlot.cc

```diff
@@ -7,9 +7,13 @@ using namespace frame_helper;
 SonarPlot::SonarPlot(QWidget *parent)
     : QFrame(parent), changedSize(true),scaleX(1),scaleY(1),range(5)
 {
+
     // apply default colormap
     applyColormap(COLORGRADIENT_JET);
 
+    numDiv = 640;
```

Comment:
base::samples::Sonar does specify the beam bearings. Read the angular difference between two beams from there (when receiving the sample), and since the visualiation requires regular bearings, verify it when reading the beams and display an error if it is not regular.